src/ml/dimensionality_reduction/pca.nim

```diff
@@ -26,3 +29,19 @@ proc pca*[T: SomeReal](x: Tensor[T], nb_components = 2): Tensor[T] =
 
   # [Nb_obs, Nb_feats] * [Nb_feats, Nb_components], don't forget to reorder component in descending order
   result = mean_centered * eigvecs[_, ^1..0|-1]
+
+  if transform:
+    return result
+  else:
+    return eigvecs
```

Comment:
We should return `eigvecs[_, ^1..0|-1]` with the principal axes in descending order (those which explain the most variance first)